### PR TITLE
fix(VTextField/VSelect): don't focus clear icon if dirty

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## How Has This Been Tested?
jest, playground

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-file-input />
    <v-text-field />
    <v-text-field clearable />
    <v-select :items="['foo']" clearable />
    <v-autocomplete :items="['foo']" clearable />
    <v-combobox :items="['foo']" clearable />
  </v-container>
</template>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VSelect/VSelect.ts [hunk 1]
```diff
@@ -419,7 +419,9 @@ export default baseMixins.extend<options>().extend({
 
       icon.children![0].data = mergeData(icon.children![0].data!, {
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSelect/VSelect.ts]: This whole thing was a mistake anyway, it should only apply to the default append icon.

File: packages/vuetify/src/components/VFileInput/__tests__/__snapshots__/VFileInput.spec.ts.snap [hunk 2]
```diff
@@ -248,11 +248,10 @@ exports[`VFileInput.ts should render 1`] = `
       </div>
       <div class="v-input__append-inner">
         <div class="v-input__icon v-input__icon--">
```
Comment (KaelWD) [inline packages/vuetify/src/components/VFileInput/__tests__/__snapshots__/VFileInput.spec.ts.snap]: icon--

File: packages/vuetify/src/components/VInput/VInput.ts [hunk 3]
```diff
@@ -201,7 +201,8 @@ export default baseMixins.extend<options>().extend({
       }
 
       return this.$createElement('div', {
-        staticClass: `v-input__icon v-input__icon--${kebabCase(type)}`,
+        staticClass: `v-input__icon`,
+        class: type ? `v-input__icon--${kebabCase(type)}` : '',
```
Comment (johnleider) [inline packages/vuetify/src/components/VInput/VInput.ts]:
```suggestion
        class: type ? `v-input__icon--${kebabCase(type)}` : undefined,
```

Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VInput/VInput.ts]: k

File: packages/vuetify/src/components/VSelect/VSelect.ts (hunk 1 above)
Comment (KaelWD) [reply packages/vuetify/src/components/VSelect/VSelect.ts]: #10489

**Pull request merged**